### PR TITLE
Adding resources, tolerations, node selectors, and affinities to the helm charts

### PR DESCRIPTION
**Pull request by lukelowery**
I have added resources, tolerations, node selectors, and affinities to the helm charts. We need these to deploy to our kubernetes cluster. I also added some default value files to charts.

### CODE REVIEW

**Comment by feng-tao** [comment]
cc @javamonkey79 Shaun, could you take a look given you have the most context on this :) ?Thanks.

**Comment by javamonkey79** [comment]
> cc @javamonkey79 Shaun, could you take a look given you have the most context on this :) ?Thanks.

Yes, I'll check it out.

File: amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml [hunk 1]
```diff
@@ -44,6 +72,10 @@ spec:
         env:
         - name: PROXY_HOST
           value: bolt://neo4j
+        {{- with .Values.metadata.esources }}
```
Comment (javamonkey79) [inline amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml]: I think this should be `resources`?

File: amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml [hunk 2]
```diff
@@ -30,12 +46,24 @@ spec:
...
     matchLabels:
       run: {{ .Chart.Name }}-{{ .Values.metadataServiceName }}
-  replicas: 1
+  replicas: {{ .Values.metadataReplicas }}
   template:
     metadata:
       labels:
         run: {{ .Chart.Name }}-{{ .Values.metadataServiceName }}
     spec:
+      {{- with default .Values.nodeSelector .Values.metadata.nodeSelector }}
```
Comment (javamonkey79) [inline amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml]: wdyt about wrapping the nodeSelector, affinity, tolerations sections in an if block, such that they are not required in all values files?

File: amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml [hunk 3]
```diff
@@ -7,12 +7,24 @@ spec:
   selector:
     matchLabels:
       run: {{ .Chart.Name }}-{{ .Values.searchServiceName }}
-  replicas: 1
+  replicas: {{ .Values.search.replicas }}
```
Comment (javamonkey79) [inline amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml]: Could you set a default here?

File: amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml (hunk 2 above)
Comment (javamonkey79) [reply amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml]: Actually, it looks like `with default` will cause this to not be used, if neither nodeSelector is defined?

File: amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml [hunk 4]
```diff
@@ -53,12 +85,24 @@ spec:
   selector:
     matchLabels:
       run: {{ .Chart.Name }}-{{ .Values.frontEndServiceName }}
-  replicas: 1
+  replicas: {{ .Values.frontEnd.replicas }}
```
Comment (javamonkey79) [inline amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml]: default here too?

File: amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml (hunk 2 above)
Comment (lukelowery) [reply amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml]: Yeah. If both are set the `metadata.nodeSelector` one overrides the `nodeSelector`. If neither are set you don't get anything in the with clause.

Comment (javamonkey79) [reply amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml]: Cool, thanks TIL

**Comment by javamonkey79** [comment]
@lukelowery FYI, also, you are not passing the CLA check...

File: amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml (hunk 3 above)
Comment (lukelowery) [reply amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml]: Alright. 

I have the default set in the values file. I think that is a better practice, but it is not a big deal.

File: amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml (hunk 1 above)
Comment (lukelowery) [reply amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml]: oops! Thanks :)

File: amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml (hunk 3 above)
Comment (javamonkey79) [reply amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml]: Yeah, I think overall having it in the values file makes sense too, but, defaulting it here means you don't have to have it in the values file... which is a little more succinct. Anyways, thanks for the change.

**Comment by javamonkey79** [comment]
@feng-tao it looks good to me. I've approved it, but, it does not seem to show as approved. I'm guessing I lack permission.

**Comment by feng-tao** [comment]
thanks @javamonkey79 for taking a look, I could do that.

@lukelowery could you configure your github email/user name locally and repush it again? This will fix the CLA check :)

**Comment by lukelowery** [comment]
@feng-tao Looks like I got my gpg keys working right now.

**Comment by feng-tao** [comment]
@lukelowery sorry for the delay(was in OOO last week). merge it now.

**Pull request merged**